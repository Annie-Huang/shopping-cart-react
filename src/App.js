import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import './App.css';
import Spinner from './components/Common/Spinner/Spinner';
import User from './components/UserList/UserList';
import ProductList from './components/ProductList/ProductList';
import ShoppingCart from './components/ShoppingCart/ShoppingCart';
import * as productActions from './actions/productActions';

export class App extends Component {

    componentDidMount = () => {
        // Need to use loadProducts in App initialization beacause product info is use in
        // both User component's pricing rules list matching as well as Product component.
        this.props.loadProducts();
    };

    render() {
        return (
            <div className='container'>
                <Spinner />
                <User />
                <ProductList />
                <ShoppingCart />
            </div>
        );
    }
}

App.propTypes = {
    loadProducts: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
    loadProducts: () => dispatch(productActions.loadProducts())
});

export default connect(null, mapDispatchToProps)(App);
