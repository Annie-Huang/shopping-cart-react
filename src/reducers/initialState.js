export default {
    products: [], // structure is: [{id: idStr, name: nameStr, price: 269.99}]
    user: {},
    ajaxCallsInProgress: 0,
    cartItems: [], // structure is: [{product: productObj, quantity: 0}]
};
