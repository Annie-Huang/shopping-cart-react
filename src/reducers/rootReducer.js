import {combineReducers} from 'redux';
import products from './productReducer';
import user from './userReducer';
import ajaxCallsInProgress from './ajaxStatusReducer';
import cartItems from './shoppingCartReducer';

const rootReducer = combineReducers({
    products,
    user,
    ajaxCallsInProgress,
    cartItems,
});

export default rootReducer;
