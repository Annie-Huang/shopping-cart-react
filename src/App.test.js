import React from 'react';
import App from './App';
import {configure as EnzymeConfig, shallow} from "enzyme/build";
import {createStore} from "redux";
import rootReducer from "./reducers/rootReducer";
import initialState from "./reducers/initialState";

EnzymeConfig({disableLifecycleMethods: true});

describe('App presentation', () => {
    const loadProducts = jest.fn();
    const props = {
        loadProducts,
    };
    const store = createStore(rootReducer, initialState);

    it('should render App component', () => {
        expect(shallow(<App {...props} />, { context: { store }})).toMatchSnapshot();
    });
});
