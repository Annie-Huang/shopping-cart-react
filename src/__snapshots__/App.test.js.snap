// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App presentation should render App component 1`] = `
<App
  loadProducts={[Function]}
/>
`;
